### PR TITLE
Remove hello world packages and documentation

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

- Remove hello world packages.
- Remove hello world documentation and update links.
- Fixed doc linter scripts to work with source files in external repositories.
- Minor fixes/restructure on documentation.

Important note:
CI will fail due to missing links. This is expected until documentation is uploaded to production and links are available.

Also, there is a [CI issue ](https://github.com/valory-xyz/open-autonomy/actions/runs/6434763721/job/17474614469?pr=2077#step:11:35) unrelated to the changes in this PR.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [x] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: docs/delegate_call.md [hunk 1]
```diff
@@ -1,3 +0,0 @@
-## Delegate call vs call
```
Comment (angrybayblade) [inline docs/delegate_call.md:1]: Why was this deleted?

File: docs/ethereum_specifics.md [hunk 2]
```diff
@@ -1,13 +0,0 @@
-## EIP-1559 Transactions
```
Comment (angrybayblade) [inline docs/ethereum_specifics.md:1]: Same question

**Review by angrybayblade — COMMENTED** [review]
This PR is a mess, so many things going on at the same time. Please try and create multistage PR next time for a task like this.

File: docs/delegate_call.md (hunk 1 above)
Comment (jmoreira-valory) [reply docs/delegate_call.md:1]: I'ts not deleted, it's moved to the FAQ, because it makes more sense there.

File: docs/ethereum_specifics.md (hunk 2 above)
Comment (jmoreira-valory) [reply docs/ethereum_specifics.md:1]: Same as above.
